packages/itwinui-react/src/core/Alert/Alert.tsx

```diff
@@ -37,73 +48,180 @@ export type AlertProps = {
 } & Omit<CommonProps, 'title'>;
 
-/**
- * A small box to quickly grab user attention and communicate a brief message
- * @example
- * <Alert>This is a basic alert.</Alert>
- * <Alert type='positive'>This is a positive alert.</Alert>
- * <Alert type='warning'>This is a warning alert.</Alert>
- * <Alert type='negative'>This is a negative alert.</Alert>
- * <Alert
- *  type='positive'
- *  clickableText="I am a clickable text"
- *  clickableTextProps={{ href: 'https://www.example.com' }}
- * >
- *   This is a positive alert with a clickable text
- * </Alert>
- */
-export const Alert = (props: AlertProps) => {
+const AlertComponent = React.forwardRef((props, ref) => {
   const {
+    as: Element = 'div',
     children,
     className,
     type = 'informational',
-    clickableText,
-    clickableTextProps,
-    onClose,
     style,
     isSticky = false,
     ...rest
   } = props;
 
   useTheme();
 
-  const StatusIcon = StatusIconMap[type];
-
   return (
-    <div
+    <Element
       className={cx('iui-alert', className)}
       data-iui-status={type}
       data-iui-variant={isSticky ? 'sticky' : undefined}
       style={style}
+      ref={ref}
+      {...rest}
+    >
+      <AlertContext.Provider value={{ type }}>{children}</AlertContext.Provider>
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'div', AlertOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.Icon component
+
+type AlertIconOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const AlertIcon = React.forwardRef((props, ref) => {
+  const { as: Element = 'svg', children, className, ...rest } = props;
+
+  const { type } = useSafeContext(AlertContext);
+
+  if (!children) {
+    const StatusIcon = StatusIconMap[type];
+    return (
+      <StatusIcon
+        className={cx('iui-alert-icon', className)}
+        ref={ref}
+        {...rest}
+      />
+    );
+  }
+
+  return (
+    <Element className={cx('iui-alert-icon', className)} ref={ref} {...rest}>
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'svg', AlertIconOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.Message component
+
+type AlertMessageOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const AlertMessage = React.forwardRef((props, ref) => {
+  const { as: Element = 'span', children, className, ...rest } = props;
+
+  return (
+    <Element className={cx('iui-alert-message', className)} ref={ref} {...rest}>
+      {children}
+    </Element>
+  );
+}) as PolymorphicForwardRefComponent<'span', AlertMessageOwnProps>;
+
+// ----------------------------------------------------------------------------
+// Alert.Action component
+
+type AlertActionOwnProps = {}; // eslint-disable-line @typescript-eslint/ban-types
+
+const AlertAction = React.forwardRef((props, ref) => {
+  const { as: Element = 'a', children, className, ...rest } = props;
```

Comment:
At first I thought this should default to `<button>` because that is safer than `<a>` (we cannot enforce passing `href`), but now I'm thinking we can detect it and change the default accordingly.

```jsx
const { as, children, className, ...rest } = props;

const Element = as ?? (!!props.href ? 'a' : 'button');
```

Reply:
This has been changed